### PR TITLE
Add mapbox

### LINKED ISSUE #72: Add map component for listing page
Should be pretty straight-forward, but this will need to include a method to inject the API key for the mapping platform from an environment variable similar to / extending #64.

I could go either way with respect to using Mapbox (consistent with other recent Exygy work) or Google Maps (consistent with SF Dahlia), and ideally we'll probably do both and let the site configuration specify. @akegan do you / SFDS have a strong preference here?

### PR DESCRIPTION
**Pull request by software-project**
Merge #64 first.
Fixes #72

### CODE REVIEW

File: shared/core/src/general.ts [hunk 1]
```diff
@@ -1,8 +1,11 @@
 export interface Address {
+  name: string
```
[the remark this answers is not in the document]
Comment (software-project) [reply shared/core/src/general.ts]: Idea here was that is name of the building/ address, so like "Triton Place".

[the remark this answers is not in the document]
Comment (bk3c) [reply shared/core/src/general.ts]: Yeah, I appreciate the intent, but would prefer not to have both listing.name and listing.buildingAddress.name etc to maintain. Also, it's one more thing you need to null check based on the usage so far. Can we get rid of this and have it be on listings only?

[the remark this answers is not in the document]
Comment (software-project) [reply shared/core/src/general.ts]: Done

**Comment by software-project** [comment]
@bk3c What about this PR? It got a bit outdated now. If you like me to I can rebase it, but please do a review first.

**Pull request merged**